pocs/camera/camera.py

```diff
@@ -275,23 +280,36 @@ def take_exposure(self,
         if not isinstance(seconds, u.Quantity):
             seconds = seconds * u.second
 
-        self.logger.debug('Taking {} second exposure on {}: {}'.format(
+        self.logger.debug('Taking {} exposure on {}: {}'.format(
             seconds, self.name, filename))
 
-        exposure_event = threading.Event()
         header = self._fits_header(seconds, dark)
 
-        self._take_exposure(seconds=seconds,
-                            filename=filename,
-                            dark=dark,
-                            exposure_event=exposure_event,
-                            header=header,
-                            *args, **kwargs)
+        if not self._exposure_event.is_set():
+            self.logger.warning('Exposure started on {} while one in progress! Waiting.'.format(
+                self))
+            self._exposure_event.wait()
+
+        # Clear event now to prevent any other exposures starting before this one is finsihed.
+        self._exposure_event.clear()
+
+        try:
+            # Camera type specific exposure set up and start
+            readout_args = self._start_exposure(seconds, filename, dark, header, *args, *kwargs)
+        except (RuntimeError, ValueError, error.PanError) as err:
+            self._exposure_event.set()
+            raise error.PanError("Error starting exposure on {}: {}".format(self, err))
+
+        # Start polling thread that will call camera type specific _readout method when done
+        readout_thread = threading.Timer(interval=get_quantity_value(seconds, unit=u.second),
```

Comment:
It doesn't need to be, no, because when the `interval` has expired polling starts. For some cameras the image will be ready to read out very soon after the length of the exposure has expired. Some not so much, but they'll just have to poll for a while.